### DISCUSSION TITLE
Where are translation files?

### DISCUSSION THREAD

**Discussion by gamg** [Laravel-Lang/common#16] [Q&A] [answered]
I tried to find the files manually (auth.php, pagination.php, passwords.php, and validation.php), but I don't see any files in any directory of your package. The Locales directory only has JSON files.

Can you help me?

**Comment by andrey-helldar** [comment, marked as answer]
In the 11th version of the project, we changed the structure in order to support several versions of projects at the same time. For example, Laravel 6, 7, 8, 9 and 10. We also improved inline translations by adding json support.

If the question is how to install them, then the answer is simple: use our [`translation manager`](https://publisher.laravel-lang.com/installation/).

If you are using Laravel 8/9 and PHP 8.1 then install the latest version. Otherwise, check the [`compatibility table`](https://publisher.laravel-lang.com/installation/compatibility.html).

**Comment by nunomira** [comment]
I'm lost...
I'm using Laravel 9. I can't find those files neither.

@andrey-helldar, you said to install "translation manager", which I did, using

`composer require laravel-lang/publisher laravel-lang/lang laravel-lang/attributes --dev`

followed by

`php artisan vendor:publish --provider="LaravelLang\Publisher\ServiceProvider"`

Still can't see the files anywhere.
And the text 

`We can't find a user with that e-mail address.`

always comes in English.

What am I missing?
Thanks!

**Reply by gamg** [reply]
did you use this command?
`php artisan lang:add xxxx`

where xxxx is the language that you want, for instance:
`php artisan lang:add fr`

**Reply by andrey-helldar** [reply]
- What version of PHP and Laravel do you have?
- Are you setting the localization in a config file or through a middleware?

![image](https://user-images.githubusercontent.com/10347617/195048924-576daf53-01c6-4603-a451-7bbf2fb1d531.png)

![image](https://user-images.githubusercontent.com/10347617/195049296-fafdf1a9-f718-49d0-98cd-0e74246b8493.png)

**Reply by nunomira** [reply]
> did you use this command? `php artisan lang:add xxxx`
> 
> where xxxx is the language that you want, for instance: `php artisan lang:add fr`

This is exactly what I was missing. Thank you very much!

**Reply by nunomira** [reply]
> * What version of PHP and Laravel do you have?
> 
>     * Are you setting the localization in a config file or through a middleware?
> 
> 
> ![image](https://user-images.githubusercontent.com/10347617/195048924-576daf53-01c6-4603-a451-7bbf2fb1d531.png)
> 
> ![image](https://user-images.githubusercontent.com/10347617/195049296-fafdf1a9-f718-49d0-98cd-0e74246b8493.png)

As soon as the files appeared, the translation worked.
Thanks!!

**Reply by andrey-helldar** [reply]
I recommend adding the `php artisan lang:update` command to the `composer.json` file.

Each time `composer update` is executed, the localization files will also be updated.

```json
"scripts": {
    "post-update-cmd": [
        "php artisan lang:update --ansi"
    ]
}
```
